### DISCUSSION TITLE
Upsert requires removing ID manually from update dict

### DISCUSSION THREAD

**Discussion by RamiAwar** [BeanieODM/beanie#334] [Question]
As part of creating a user, i'm using upsert which would allow me to neatly cover the case of 1- user does not exist and 2- user exists

I want to update all the user details in case they exist, except for the ID obviously.

But in order to do that, I need to pass in all the properties except for the ID in the upsert update section `Set(user_dict)`.

Doesn't it make sense for Beanie to take care of ignoring the ID in the case of updating inside upsert? 
Passing it in with the ID results in a serialization error on creation, and in a NULL ID on the database document which is not good. That's why I need to create a dict without the ID here.

```python
    # Check if user exists using twitter user ID
    # This needs to change when adding other platforms support
    user_dict = user.dict(exclude={"id"})
    await User.find_one(User.twitter_user_id == user.twitter_user_id).upsert(
        Set(user_dict),
        on_insert=user,
    )
```

**Comment by roman-right** [comment]
Hm. I didn't think about this case, but it looks like it needs some syntax sugar like set_or_insert, yup. Or just ignore the id there. I have to think about all the corner cases. I'll add this to my todo. Thank you!
